rust/src/request.rs

```diff
@@ -1,30 +1,45 @@
 use crate::{
     client::ClientError,
     crypto::ByteObject,
+    mask::Model,
     request::Proxy::{InMem, Remote},
-    service::{data::RoundParametersData, Handle},
-    ParticipantPublicKey,
+    services::{Fetcher, PetMessageHandler},
+    state_machine::coordinator::RoundParameters,
+    SeedDict,
     SumDict,
+    SumParticipantPublicKey,
     UpdateSeedDict,
 };
 use bytes::Bytes;
 use reqwest::{Client, Error, IntoUrl, Response, StatusCode};
 
-#[derive(Debug)]
 /// Proxy for communicating with the service.
 pub enum Proxy {
-    InMem(Handle),
+    InMem(
+        Box<dyn Fetcher + Send + Sync>,
+        Box<dyn PetMessageHandler + Send + Sync>,
+    ),
```

Comment:
from looking at `rest.rs` where the traits are used as simple trait bounds, it seems that we may get rid of dynamic dispatch here as well. this is not urgent though, since the in-memory client is used for testing only.

Reply:
Agree. This is a "quick & dirty" solution